File: 2015/Day_01/Python/part_2.py
import sys
sys.path.insert(0, '../../input_parsing')
import parse_input


def parse_floor_directions(instructions: str) -> int:
    """Take a series of parenthesis and figure out what floor that leaves Santa at."""
    floor = 0
    for position, parenthesis in enumerate(instructions, start=1):
        if parenthesis == "(":
            floor += 1
        elif parenthesis == ")":
            floor -= 1
        if floor == -1:
            return position


if __name__ == "__main__":
    floor_directions = parse_input.input_per_line('../input.txt')
    print(parse_floor_directions(floor_directions[0]))

File: 2015/Day_01/Ruby/part_2.rb
require "../../input_parsing/parse_input"

def parse_directions(parenthesis)
    floor = 0
    position = 1
    parenthesis.each do |paren|
        if paren == "("
            floor += 1
        elsif paren == ")"
            floor -= 1
        end
        if floor == -1
            return position
        end
        position += 1
        end
end



input_text = input_per_line('../input.txt')
puts parse_directions(input_text[0].split(//))
